### PR TITLE
Create AVL Tree.md

### PR DESCRIPTION
**Pull request by vishantrathi**
This pull request introduces [briefly describe the main changes, e.g., "an implementation of AVL trees in Python, including insertion and balancing operations"]. The goal of this PR is to [explain the purpose, e.g., "enhance the data structure library with a self-balancing tree that ensures efficient search, insertion, and deletion operations"].

Changes Made
Implemented AVL Tree Class: Created a new AVLTree class with methods for insertion, rotation, and height management. Node Structure: Defined a Node class to represent each node in the AVL tree, including properties for the key, left and right children, and height. Balancing Logic: Added logic to maintain the balance of the tree during insertions, including left and right rotations. Unit Tests: Included unit tests to verify the correctness of the AVL tree operations.

### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue number)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Trees/AVL Tree.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        return y
+
+    def left_rotate(self, z):
+        y = z.right
+        T2 = y.left
+        y.left = z
+        z.right = T2
+        z.height = 1 + max(self.get_height(z.left), self.get_height(z.right))
+        y.height = 1 + max(self.get_height(y.left), self.get_height(y.right))
+        return y
```
Comment (ajay-dhangar) [inline docs/Trees/AVL Tree.md:96]: Close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Review by vishantrathi — COMMENTED** [review]
changes done as requested by the maintainer

File: docs/Trees/AVL Tree.md (hunk 1 above)
Comment (vishantrathi) [reply docs/Trees/AVL Tree.md:96]: Done as you told 
Please merge after review.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**